File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/CustomerTaxData.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;

/**
 * Summarized tax data for a customer returned when using the customer mode API.
 */
public class CustomerTaxData {

    @SerializedName("PCode")
    private Integer pCode = null;

    @SerializedName("TaxType")
    private Integer taxType = null;

    @SerializedName("TaxLevel")
    private Integer taxLevel = null;

    @SerializedName("CalculationType")
    private Integer calculationType = null;

    @SerializedName("Rate")
    private Double rate = null;

    @SerializedName("TaxAmount")
    private Double taxAmount = null;

    @SerializedName("ExemptSaleAmount")
    private Double exemptSaleAmount = null;

    @SerializedName("Description")
    private String description = null;

    @SerializedName("Surcharge")
    private Integer surcharge = null;

    @SerializedName("MaxBase")
    private Double maxBase = null;

    @SerializedName("MinBase")
    private Double minBase = null;

    @SerializedName("ExcessTax")
    private Double excessTax = null;

    @SerializedName("TotalCharge")
    private Double totalCharge = null;

    @SerializedName("CategoryID")
    private Integer categoryID = null;

    @SerializedName("CategoryDescription")
    private String categoryDescription = null;

    @SerializedName("Lines")
    private Integer lines = null;

    @SerializedName("Locations")
    private Integer locations = null;

    @SerializedName("Minutes")
    private Double minutes = null;

    /**
     * Sets the PCode for the reporting jurisdiction.
     * @param pCode
     * @return
     */
    public CustomerTaxData pCode(Integer pCode) {
        this.pCode = pCode;
        return this;
    }

    /**
     * PCode for the reporting jurisdiction.
     * @return pCode
     */
    public Integer getPCode() {
        return pCode;
    }

    /**
     * Sets the PCode for the reporting jurisdiction.
     * @param pCode
     */
    public void setPCode(Integer pCode) {
        this.pCode = pCode;
    }

    /**
     * Sets the tax type. See the CommsPlatform.API.WrapperClasses User's Manual
     * for a list of valid tax types.
     * @param taxType
     * @return
     */
    public CustomerTaxData taxType(Integer taxType) {
        this.taxType = taxType;
        return this;
    }

    /**
     * Tax type. See the CommsPlatform.API.WrapperClasses User's Manual for a
     * list of valid tax types.
     * @return taxType
     */
    public Integer getTaxType() {
        return taxType;
    }

    /**
     * Sets the tax type. See the CommsPlatform.API.WrapperClasses User's Manual
     * for a list of valid tax types.
     * @param taxType
     */
    public void setTaxType(Integer taxType) {
        this.taxType = taxType;
    }

    /**
     * Sets the tax level. See the CommsPlatform.API.WrapperClasses User's
     * Manual for a list of valid tax levels.
     * @param taxLevel
     * @return
     */
    public CustomerTaxData taxLevel(Integer taxLevel) {
        this.taxLevel = taxLevel;
        return this;
    }

    /**
     * Tax level. See the CommsPlatform.API.WrapperClasses User's Manual for a
     * list of valid tax levels.
     * @return taxLevel
     */
    public Integer getTaxLevel() {
        return taxLevel;
    }

    /**
     * Sets the tax level. See the CommsPlatform.API.WrapperClasses User's
     * Manual for a list of valid tax levels.
     * @param taxLevel
     */
    public void setTaxLevel(Integer taxLevel) {
        this.taxLevel = taxLevel;
    }

    /**
     * Sets the calculation type. See the CommsPlatform.API.WrapperClasses
     * User's Manual for a list of valid calculation types.
     * @param calculationType
     * @return
     */
    public CustomerTaxData calculationType(Integer calculationType) {
        this.calculationType = calculationType;
        return this;
    }

    /**
     * Calculation type. See the CommsPlatform.API.WrapperClasses User's Manual
     * for a list of valid calculation types.
     * @return calculationType
     */
    public Integer getCalculationType() {
        return calculationType;
    }

    /**
     * Sets the calculation type. See the CommsPlatform.API.WrapperClasses
     * User's Manual for a list of valid calculation types.
     * @param calculationType
     */
    public void setCalculationType(Integer calculationType) {
        this.calculationType = calculationType;
    }

    /**
     * Sets the tax rate.
     * @param rate
     * @return
     */
    public CustomerTaxData rate(Double rate) {
        this.rate = rate;
        return this;
    }

    /**
     * Tax rate.
     * @return rate
     */
    public Double getRate() {
        return rate;
    }

    /**
     * Sets the tax rate.
     * @param rate
     */
    public void setRate(Double rate) {
        this.rate = rate;
    }

    /**
     * Sets the tax amount.
     * @param taxAmount
     * @return
     */
    public CustomerTaxData taxAmount(Double taxAmount) {
        this.taxAmount = taxAmount;
        return this;
    }

    /**
     * Tax amount.
     * @return taxAmount
     */
    public Double getTaxAmount() {
        return taxAmount;
    }

    /**
     * Sets the tax amount.
     * @param taxAmount
     */
    public void setTaxAmount(Double taxAmount) {
        this.taxAmount = taxAmount;
    }

    /**
     * Sets the amount of the charge exempt from taxes.
     * @param exemptSaleAmount
     * @return
     */
    public CustomerTaxData exemptSaleAmount(Double exemptSaleAmount) {
        this.exemptSaleAmount = exemptSaleAmount;
        return this;
    }

    /**
     * Amount of the charge exempt from taxes.
     * @return exemptSaleAmount
     */
    public Double getExemptSaleAmount() {
        return exemptSaleAmount;
    }

    /**
     * Sets the amount of the charge exempt from taxes.
     * @param exemptSaleAmount
     */
    public void setExemptSaleAmount(Double exemptSaleAmount) {
        this.exemptSaleAmount = exemptSaleAmount;
    }

    /**
     * Sets the tax description.
     * @param description
     * @return
     */
    public CustomerTaxData description(String description) {
        this.description = description;
        return this;
    }

    /**
     * Tax description.
     * @return description
     */
    public String getDescription() {
        return description;
    }

    /**
     * Sets the tax description.
     * @param description
     */
    public void setDescription(String description) {
        this.description = description;
    }

    /**
     * Sets a value indicating if tax is considered a surcharge.
     * @param surcharge
     * @return
     */
    public CustomerTaxData surcharge(Integer surcharge) {
        this.surcharge = surcharge;
        return this;
    }

    /**
     * Indicates if tax is considered a surcharge.
     * @return surcharge
     */
    public Integer getSurcharge() {
        return surcharge;
    }

    /**
     * Sets a value indicating if tax is considered a surcharge.
     * @param surcharge
     */
    public void setSurcharge(Integer surcharge) {
        this.surcharge = surcharge;
    }

    /**
     * Sets the max amount to which tax is applied.
     * @param maxBase
     * @return
     */
    public CustomerTaxData maxBase(Double maxBase) {
        this.maxBase = maxBase;
        return this;
    }

    /**
     * Max amount to which tax is applied.
     * @return maxBase
     */
    public Double getMaxBase() {
        return maxBase;
    }

    /**
     * Sets the max amount to which tax is applied.
     * @param maxBase
     */
    public void setMaxBase(Double maxBase) {
        this.maxBase = maxBase;
    }

    /**
     * Sets the min amount to which tax is applied.
     * @param minBase
     * @return
     */
    public CustomerTaxData minBase(Double minBase) {
        this.minBase = minBase;
        return this;
    }

    /**
     * Min amount to which tax is applied.
     * @return minBase
     */
    public Double getMinBase() {
        return minBase;
    }

    /**
     * Sets the min amount to which tax is applied.
     * @param minBase
     */
    public void setMinBase(Double minBase) {
        this.minBase = minBase;
    }

    /**
     * Sets the rate for amount above max base.
     * @param excessTax
     * @return
     */
    public CustomerTaxData excessTax(Double excessTax) {
        this.excessTax = excessTax;
        return this;
    }

    /**
     * Rate for amount above max base.
     * @return excessTax
     */
    public Double getExcessTax() {
        return excessTax;
    }

    /**
     * Sets the rate for amount above max base.
     * @param excessTax
     */
    public void setExcessTax(Double excessTax) {
        this.excessTax = excessTax;
    }

    /**
     * Sets the total charge for this tax.
     * @param totalCharge
     * @return
     */
    public CustomerTaxData totalCharge(Double totalCharge) {
        this.totalCharge = totalCharge;
        return this;
    }

    /**
     * Sum of charges for this customer.
     * @return totalCharge
     */
    public Double getTotalCharge() {
        return totalCharge;
    }

    /**
     * Sets the total charge for this tax.
     * @param totalCharge
     */
    public void setTotalCharge(Double totalCharge) {
        this.totalCharge = totalCharge;
    }

    /**
     * Sets the tax category ID.
     * @param categoryID
     * @return
     */
    public CustomerTaxData categoryID(Integer categoryID) {
        this.categoryID = categoryID;
        return this;
    }

    /**
     * Tax category ID.
     * @return categoryID
     */
    public Integer getCategoryID() {
        return categoryID;
    }

    /**
     * Sets the tax category ID.
     * @param categoryID
     */
    public void setCategoryID(Integer categoryID) {
        this.categoryID = categoryID;
    }

    /**
     * Sets the tax category description.
     * @param categoryDescription
     * @return
     */
    public CustomerTaxData categoryDescription(String categoryDescription) {
        this.categoryDescription = categoryDescription;
        return this;
    }

    /**
     * Tax category description.
     * @return categoryDescription
     */
    public String getCategoryDescription() {
        return categoryDescription;
    }

    /**
     * Sets the tax category description.
     * @param categoryDescription
     */
    public void setCategoryDescription(String categoryDescription) {
        this.categoryDescription = categoryDescription;
    }

    /**
     * Sets the number of lines used to calculate the tax (for per-line tax
     * calculations).
     * @param lines
     * @return
     */
    public CustomerTaxData lines(Integer lines) {
        this.lines = lines;
        return this;
    }

    /**
     * Number of lines used to calculate the tax (for per-line tax
     * calculations).
     * @return lines
     */
    public Integer getLines() {
        return lines;
    }

    /**
     * Sets the number of lines used to calculate the tax (for per-line tax
     * calculations).
     * @param lines
     */
    public void setLines(Integer lines) {
        this.lines = lines;
    }

    /**
     * Sets the number of locations.
     * @param locations
     * @return
     */
    public CustomerTaxData locations(Integer locations) {
        this.locations = locations;
        return this;
    }

    /**
     * Sum of locations from customer input.
     * @return locations
     */
    public Integer getLocations() {
        return locations;
    }

    /**
     * Sets the number of locations.
     * @param locations
     */
    public void setLocations(Integer locations) {
        this.locations = locations;
    }

    /**
     * Sets the number of minutes used to calculate the tax (for per-minute tax
     * calculations).
     * @param minutes
     * @return
     */
    public CustomerTaxData minutes(Double minutes) {
        this.minutes = minutes;
        return this;
    }

    /**
     * Number of minutes used to calculate the tax (for per-minute tax
     * calculations).
     * @return minutes
     */
    public Double getMinutes() {
        return minutes;
    }

    /**
     * Sets the number of minutes used to calculate the tax (for per-minute tax
     * calculations).
     * @param minutes
     */
    public void setMinutes(Double minutes) {
        this.minutes = minutes;
    }
}

File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/CustomerTaxData.js
/**
 * Summarized tax data for a customer returned when using the customer mode API.
 */

function CustomerTaxData() {

    this.PCode = 0;
    this.TaxType = 0;
    this.TaxLevel = 0;
    this.CalculationType = 0;
    this.Rate = 0;
    this.TaxAmount = 0;
    this.ExemptSaleAmount = 0;
    this.Description = "";
    this.Surcharge = 0;
    this.MaxBase = 0;
    this.MinBase = 0;
    this.ExcessTax = 0;
    this.TotalCharge = 0;
    this.CategoryID = 0;
    this.CategoryDescription = "";
    this.Lines = 0;
    this.Locations = 0;
    this.Minutes = 0;

    /**
     * Sets the PCode for the reporting jurisdiction.
     * @param pCode Number
     * @return Reference to this object for method chaining.
     */
    this.pCode = function (pCode) {
        this.PCode = pCode;
        return this;
    }

    /**
     * PCode for the reporting jurisdiction.
     * @return Number
     */
    this.getPCode = function () {
        return this.PCode;
    }

    /* Reference to this object for method chaining.*
     * Sets the PCode for the reporting jurisdiction.
     * @param pCode Number
     */
    this.setPCode = function (pCode) {
        this.PCode = pCode;
    }

    /**
     * Sets the tax type. See the CommsPlatform.API.WrapperClasses User's Manual
     * for a list of valid tax types.
     * @param taxType Number
     * @return Reference to this object for method chaining.
     */
    this.taxType = function (taxType) {
        this.TaxType = taxType;
        return this;
    }

    /**
     * Tax type. See the CommsPlatform.API.WrapperClasses User's Manual for a
     * list of valid tax types.
     * @return Number
     */
    this.getTaxType = function () {
        return this.TaxType;
    }

    /* Reference to this object for method chaining.*
     * Sets the tax type. See the CommsPlatform.API.WrapperClasses User's Manual
     * for a list of valid tax types.
     * @param taxType Number
     */
    this.setTaxType = function (taxType) {
        this.TaxType = taxType;
    }

    /**
     * Sets the tax level. See the CommsPlatform.API.WrapperClasses User's
     * Manual for a list of valid tax levels.
     * @param taxLevel Number
     * @return Reference to this object for method chaining.
     */
    this.taxLevel = function (taxLevel) {
        this.TaxLevel = taxLevel;
        return this;
    }

    /**
     * Tax level. See the CommsPlatform.API.WrapperClasses User's Manual for a
     * list of valid tax levels.
     * @return Number
     */
    this.getTaxLevel = function () {
        return this.TaxLevel;
    }

    /* Reference to this object for method chaining.*
     * Sets the tax level. See the CommsPlatform.API.WrapperClasses User's
     * Manual for a list of valid tax levels.
     * @param taxLevel Number
     */
    this.setTaxLevel = function (taxLevel) {
        this.TaxLevel = taxLevel;
    }

    /**
     * Sets the calculation type. See the CommsPlatform.API.WrapperClasses
     * User's Manual for a list of valid calculation types.
     * @param calculationType Number
     * @return Reference to this object for method chaining.
     */
    this.calculationType = function (calculationType) {
        this.CalculationType = calculationType;
        return this;
    }

    /**
     * Calculation type. See the CommsPlatform.API.WrapperClasses User's Manual
     * for a list of valid calculation types.
     * @return Number
     */
    this.getCalculationType = function () {
        return this.CalculationType;
    }

    /* Reference to this object for method chaining.*
     * Sets the calculation type. See the CommsPlatform.API.WrapperClasses
     * User's Manual for a list of valid calculation types.
     * @param calculationType Number
     */
    this.setCalculationType = function (calculationType) {
        this.CalculationType = calculationType;
    }

    /**
     * Sets the tax rate.
     * @param rate Number
     * @return Reference to this object for method chaining.
     */
    this.rate = function (rate) {
        this.Rate = rate;
        return this;
    }

    /**
     * Tax rate.
     * @return Number
     */
    this.getRate = function () {
        return this.Rate;
    }

    /* Reference to this object for method chaining.*
     * Sets the tax rate.
     * @param rate Number
     */
    this.setRate = function (rate) {
        this.Rate = rate;
    }

    /**
     * Sets the tax amount.
     * @param taxAmount Number
     * @return Reference to this object for method chaining.
     */
    this.taxAmount = function (taxAmount) {
        this.TaxAmount = taxAmount;
        return this;
    }

    /**
     * Tax amount.
     * @return Number
     */
    this.getTaxAmount = function () {
        return this.TaxAmount;
    }

    /* Reference to this object for method chaining.*
     * Sets the tax amount.
     * @param taxAmount Number
     */
    this.setTaxAmount = function (taxAmount) {
        this.TaxAmount = taxAmount;
    }

    /**
     * Sets the amount of the charge exempt from taxes.
     * @param exemptSaleAmount Number
     * @return Reference to this object for method chaining.
     */
    this.exemptSaleAmount = function (exemptSaleAmount) {
        this.ExemptSaleAmount = exemptSaleAmount;
        return this;
    }

    /**
     * Amount of the charge exempt from taxes.
     * @return Number
     */
    this.getExemptSaleAmount = function () {
        return this.ExemptSaleAmount;
    }

    /* Reference to this object for method chaining.*
     * Sets the amount of the charge exempt from taxes.
     * @param exemptSaleAmount Number
     */
    this.setExemptSaleAmount = function (exemptSaleAmount) {
        this.ExemptSaleAmount = exemptSaleAmount;
    }

    /**
     * Sets the tax description.
     * @param description String
     * @return Reference to this object for method chaining.
     */
    this.description = function (description) {
        this.Description = description;
        return this;
    }

    /**
     * Tax description.
     * @return String
     */
    this.getDescription = function () {
        return this.Description;
    }

    /* Reference to this object for method chaining.*
     * Sets the tax description.
     * @param description String
     */
    this.setDescription = function (description) {
        this.Description = description;
    }

    /**
     * Sets a value indicating if tax is considered a surcharge.
     * @param surcharge Number
     * @return Reference to this object for method chaining.
     */
    this.surcharge = function (surcharge) {
        this.Surcharge = surcharge;
        return this;
    }

    /**
     * Indicates if tax is considered a surcharge.
     * @return Number
     */
    this.getSurcharge = function () {
        return this.Surcharge;
    }

    /* Reference to this object for method chaining.*
     * Sets a value indicating if tax is considered a surcharge.
     * @param surcharge Number
     */
    this.setSurcharge = function (surcharge) {
        this.Surcharge = surcharge;
    }

    /**
     * Sets the max amount to which tax is applied.
     * @param maxBase Number
     * @return Reference to this object for method chaining.
     */
    this.maxBase = function (maxBase) {
        this.MaxBase = maxBase;
        return this;
    }

    /**
     * Max amount to which tax is applied.
     * @return Number
     */
    this.getMaxBase = function () {
        return this.MaxBase;
    }

    /* Reference to this object for method chaining.*
     * Sets the max amount to which tax is applied.
     * @param maxBase Number
     */
    this.setMaxBase = function (maxBase) {
        this.MaxBase = maxBase;
    }

    /**
     * Sets the min amount to which tax is applied.
     * @param minBase Number
     * @return Reference to this object for method chaining.
     */
    this.minBase = function (minBase) {
        this.MinBase = minBase;
        return this;
    }

    /**
     * Min amount to which tax is applied.
     * @return Number
     */
    this.getMinBase = function () {
        return this.MinBase;
    }

    /* Reference to this object for method chaining.*
     * Sets the min amount to which tax is applied.
     * @param minBase Number
     */
    this.setMinBase = function (minBase) {
        this.MinBase = minBase;
    }

    /**
     * Sets the rate for amount above max base.
     * @param excessTax Number
     * @return Reference to this object for method chaining.
     */
    this.excessTax = function (excessTax) {
        this.ExcessTax = excessTax;
        return this;
    }

    /**
     * Rate for amount above max base.
     * @return Number
     */
    this.getExcessTax = function () {
        return this.ExcessTax;
    }

    /* Reference to this object for method chaining.*
     * Sets the rate for amount above max base.
     * @param excessTax Number
     */
    this.setExcessTax = function (excessTax) {
        this.ExcessTax = excessTax;
    }

    /**
     * Sets the total charge for this tax.
     * @param totalCharge Number
     * @return Reference to this object for method chaining.
     */
    this.totalCharge = function (totalCharge) {
        this.TotalCharge = totalCharge;
        return this;
    }

    /**
     * Sum of charges for this customer.
     * @return Number
     */
    this.getTotalCharge = function () {
        return this.TotalCharge;
    }

    /* Reference to this object for method chaining.*
     * Sets the total charge for this tax.
     * @param totalCharge Number
     */
    this.setTotalCharge = function (totalCharge) {
        this.TotalCharge = totalCharge;
    }

    /**
     * Sets the tax category ID.
     * @param categoryID Number
     * @return Reference to this object for method chaining.
     */
    this.categoryID = function (categoryID) {
        this.CategoryID = categoryID;
        return this;
    }

    /**
     * Tax category ID.
     * @return Number
     */
    this.getCategoryID = function () {
        return this.CategoryID;
    }

    /* Reference to this object for method chaining.*
     * Sets the tax category ID.
     * @param categoryID Number
     */
    this.setCategoryID = function (categoryID) {
        this.CategoryID = categoryID;
    }

    /**
     * Sets the tax category description.
     * @param categoryDescription String
     * @return Reference to this object for method chaining.
     */
    this.categoryDescription = function (categoryDescription) {
        this.CategoryDescription = categoryDescription;
        return this;
    }

    /**
     * Tax category description.
     * @return String
     */
    this.getCategoryDescription = function () {
        return this.CategoryDescription;
    }

    /* Reference to this object for method chaining.*
     * Sets the tax category description.
     * @param categoryDescription String
     */
    this.setCategoryDescription = function (categoryDescription) {
        this.CategoryDescription = categoryDescription;
    }

    /**
     * Sets the number of lines used to calculate the tax (for per-line tax
     * calculations).
     * @param lines Number
     * @return Reference to this object for method chaining.
     */
    this.lines = function (lines) {
        this.Lines = lines;
        return this;
    }

    /**
     * Number of lines used to calculate the tax (for per-line tax
     * calculations).
     * @return Number
     */
    this.getLines = function () {
        return this.Lines;
    }

    /* Reference to this object for method chaining.*
     * Sets the number of lines used to calculate the tax (for per-line tax
     * calculations).
     * @param lines Number
     */
    this.setLines = function (lines) {
        this.Lines = lines;
    }

    /**
     * Sets the number of locations.
     * @param locations Number
     * @return Reference to this object for method chaining.
     */
    this.locations = function (locations) {
        this.Locations = locations;
        return this;
    }

    /**
     * Sum of locations from customer input.
     * @return Number
     */
    this.getLocations = function () {
        return this.Locations;
    }

    /* Reference to this object for method chaining.*
     * Sets the number of locations.
     * @param locations Number
     */
    this.setLocations = function (locations) {
        this.Locations = locations;
    }

    /**
     * Sets the number of minutes used to calculate the tax (for per-minute tax
     * calculations).
     * @param minutes Number
     * @return Reference to this object for method chaining.
     */
    this.minutes = function (minutes) {
        this.Minutes = minutes;
        return this;
    }

    /**
     * Number of minutes used to calculate the tax (for per-minute tax
     * calculations).
     * @return Number
     */
    this.getMinutes = function () {
        return this.Minutes;
    }

    /* Reference to this object for method chaining.*
     * Sets the number of minutes used to calculate the tax (for per-minute tax
     * calculations).
     * @param minutes Number
     */
    this.setMinutes = function (minutes) {
        this.Minutes = minutes;
    }
}
